### DISCUSSION TITLE
EV3: wrong time on brick

### DISCUSSION THREAD

**Discussion by hagbeck** [pybricks/support#494] [Q&A] [answered]
Hi,
I'm using `time.localtime(time.time())` on my EV3 brick and getting wrong date and time (Bluetooth and WIFI enabled and connected). 
If go directly to the bricks terminal I can reproduce this via timedatectl (startet at Fri, 24 Sep 2021 10:51:30 +0200).

```
robot@ev3dev:~$ timedatectl status 
      Local time: Mon 2021-09-20 23:22:23 UTC
  Universal time: Mon 2021-09-20 23:22:23 UTC
        RTC time: n/a
       Time zone: Etc/UTC (UTC, +0000)
 Network time on: yes
 NTP synchronized: yes
 RTC in local TZ: no
```

I understand, the time should be updated by NTP. But it doesn't. What's going wrong here?

Any ideas?

**Comment by dlech** [comment]
There is no battery backup for the clock, so when the EV3 is turned off, the clock stops ticking. At boot, it restores the last known time (`fake-hwclock`). NTP will correct the time, but only if the EV3 is connected to the Internet. The fact that the date is close to the present tells me that NTP was working a few days ago or that you manually set the date.

**Comment by hagbeck** [comment, marked as answer]
Based on your explanations I experimented a bit more and asserted that there is a kind of winner and loser network. I have configured Bluetooth and WiFi at startup of the brick via auto connect. If my PC is online before I start the brick, the Bluetooth network wins and the brick gets an 10... IP address.
If I start the brick before my PC starts, the WiFi wins and the brick gets an 192.... IP address. In the second case the NTP updates the clock directly.

If I restart the WiFi on the brick in the first case, the WiFi beats the Bluetooth and the clock is updated.

Because I realized now, that it is currently not necessary to auto connect both networks, I selected WiFi and the clock is fine.

Thanks for your inspiration! :-)
